File: library/src/test/java/com/proxerme/library/api/DateAdapterTest.java
package com.proxerme.library.api;

import org.junit.Before;
import org.junit.Test;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatExceptionOfType;

/**
 * TODO: Describe class
 *
 * @author Ruben Gees
 */
public class DateAdapterTest {

    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd hh:mm:ss");

    private DateAdapter adapter;

    @Before
    public void setUp() throws Exception {
        adapter = new DateAdapter();
    }

    @Test
    public void testFromJsonTimestamp() throws Exception {
        assertThat(adapter.fromJson("12345")).isEqualTo(new Date(12345 * 1000));
    }

    @Test
    public void testFromJsonIso() throws Exception {
        assertThat(adapter.fromJson("2010-01-01 23:12:10"))
                .isEqualTo(DATE_FORMAT.parse("2010-01-01 23:12:10"));
    }

    @Test
    public void testFromJsonMalformed() throws Exception {
        assertThatExceptionOfType(ParseException.class).isThrownBy(() -> adapter.fromJson("malformed"));
    }

    @Test
    public void testToJson() throws Exception {
        assertThat(adapter.toJson(new Date(123))).isEqualTo(123);
    }
}

File: library/src/test/kotlin/me/proxer/library/internal/adapter/DateAdapterTest.kt
package me.proxer.library.internal.adapter

import me.proxer.library.toProxerDate
import me.proxer.library.toProxerDateTime
import org.amshove.kluent.invoking
import org.amshove.kluent.shouldEqual
import org.amshove.kluent.shouldThrow
import org.junit.jupiter.api.Test
import java.text.ParseException
import java.util.Date

/**
 * @author Ruben Gees
 */
class DateAdapterTest {

    private val adapter = DateAdapter()

    @Test
    fun testFromJsonTimestamp() {
        adapter.fromJson("12345") shouldEqual Date((12345 * 1000).toLong())
    }

    @Test
    fun testFromJsonIso() {
        adapter.fromJson("2010-01-01 23:12:10") shouldEqual "2010-01-01 23:12:10".toProxerDateTime()
    }

    @Test
    fun testFromJsonIsoNoTime() {
        adapter.fromJson("2010-01-01") shouldEqual "2010-01-01".toProxerDate()
    }

    @Test
    fun testFromJsonIsoNoTimeEmpty() {
        adapter.fromJson("0000-00-00") shouldEqual "0000-00-00".toProxerDate()
    }

    @Test
    fun testFromJsonMalformed() {
        invoking { adapter.fromJson("malformed") } shouldThrow ParseException::class
    }

    @Test
    fun testToJson() {
        adapter.toJson(Date(123)) shouldEqual 123
    }
}
